Resources/public/less/layouts.less

```diff
@@ -1,338 +1,339 @@
+    top: 9px;
+    left: 8px;
+    z-index: 1;
+    .sprite();
+  }
+  input.search-query {
+    padding-left: 25px;
+    width: 100%;
+    .placeholder(@blue);
+    &:focus{
+      .placeholder(red);
     }
+  }
 }
 
 .navbar.extra-navi .nav > li#login {
+  a.hide-login-form {
+    display: none;
+  }
+  a.show-login-form {
+    display: block;
+  }
+  &:target {
+    .login-form {
+      .box-shadow(0 5px 5px rgba(0,0,0,0.35));
+      background: @blue;
+      #gradient > .vertical(@blue, @blueDark);
+    }
+  }
+  .login-form {
+    width: 280px;
+    margin: 0;
+    color: white;
+    position: absolute;
+    right: 0;
+    z-index: 10;
+    height: 0px;
+    overflow: hidden;
+    bottom: 0;
+    .border-radius(3px 0 3px 3px);
+    .transition(all 0.2s ease);
+    fieldset {
+      display: block;
+      padding: 10px;
+      a.forgot-password, a.forgot-password:hover {
+        color: white;
+        display: inline-block;
+        margin-top: 14px;
+      }
+      label {
+        display: block;
+        margin-bottom: 9px;
+        input {
+          width: 100%;
+          color: @textColor;
+          font-weight: normal;
+          font-size: 15px;
+        }
+      }
+    }
+    a.close:after{
+      content: "x";
+    }
+  }
+  &:target{
+    background-color: @blue;
+    > a,
+    > a:hover {
+      color: white;
+    }
     a.hide-login-form {
-        display: none;
+      display: block;
     }
     a.show-login-form {
-        display: block;
+      display: none;
     }
     .login-form {
-        margin: 0;
-        background: @blue;
-        color: white;
-        position: absolute;
-        bottom: -140px;
-        right: 0;
-        z-index: 10;
-        height: 0px;
-        overflow: hidden;
-        bottom: 0;
-        #gradient > .vertical(@blue, @blueDark);
-        .border-radius(3px 0 3px 3px);
-        .box-shadow(0 5px 5px rgba(0,0,0,0.35));
-        .transition(all 0.2s ease);
-        fieldset {
-            display: block;
-            padding: 10px;
-            a.forgot-password,
-            a.forgot-password:hover {
-                color: white;
-                display: inline-block;
-                margin-top: 18px;
-            }
-            label {
-                display: block;
-                padding-right: 10px;
-                input {
-                    width: 100%;
-                }
-            }
-        }
-        a.close:after{
-            content: "x";
-        }
-    }
-    &:target{
-        background-color: @blue;
-        > a,
-        > a:hover {
-            color: white;
-        }
-        a.hide-login-form {
-            display: block;
-        }
-        a.show-login-form {
-            display: none;
-        }
-        .login-form {
-            display: block;
-            height: 140px;
-            overflow: hidden;
-            bottom: -140px;
-        }
+      display: block;
+      height: 130px;
+      overflow: hidden;
+      bottom: -130px;
```

Comment:
Nitpick: it can be written like that: `transform: translateY(100%);`. You won't have to care about the element's height value.